tests/testthat/helper-functions.R

```diff
@@ -417,3 +417,11 @@ find_test_path <- function(dir = ".") {
   file.path(dir, ext_dir)
 }
 
+runtime_evaluate <- function(script=NULL,return.value=FALSE){
+  if (return.value){
+    value<- remDr$Runtime$evaluate(script,returnByValue = TRUE)$result$value
```

Comment:
please avoid repeating `remDr$Runtime$evaluate(script,returnByValue = TRUE)` for example you can do as below
```r
eval.result <-  remDr$Runtime$evaluate(script,returnByValue = TRUE)
if(return.value)eval.result$result$value else eval.result
```

Reply:
in general whenever you have a repeated block of code, please use for/if/etc to avoid

Reply:
@tdhock I made the change and did a commit. but for last 3 commits I am facing issues on GitHub Actions. The tests are passing locally. For last two commits, checks never stopped despite many hours. So I had to stop them manually.

Reply:
> @tdhock I made the change and did a commit. but for last 3 commits I am facing issues on GitHub Actions. The tests are passing locally. For last two commits, checks never stopped despite many hours. So I had to stop them manually. And I got these errors:
> 
> [8623:8623:0713/031337.057125:ERROR:ozone_platform_x11.cc(244)] Missing X server or $DISPLAY [8623:8623:0713/031337.057164:ERROR:env.cc(258)] The platform failed to initialize. Exiting. Error: The operation was canceled.
> 
> It's some issue on GitHub actions specifically.

solved it by reverting change of removing extra package

Reply:
this does not make sense to me.
You re-added the extra-packages line, and you think that fixed the tests? Where do you see that? and why do you think adding extra-packages line would fix the test failures? 

First of all I don't see any passing test results. I see neither check mark nor red X by most recent commit "reverted changes" This suggests there is a syntax error in yaml file.

Also I think the extra-packages line is not doing anything, because that step is already installing all of those packages since they are listed in DESCRIPTION.

Reply:
> this does not make sense to me. You re-added the extra-packages line, and you think that fixed the tests? Where do you see that? and why do you think adding extra-packages line would fix the test failures?
> 
> First of all I don't see any passing test results. I see neither check mark nor red X by most recent commit "reverted changes" This suggests there is a syntax error in yaml file.
> 
> Also I think the extra-packages line is not doing anything, because that step is already installing all of those packages since they are listed in DESCRIPTION.

@tdhock I was confused but still thought that the checks have probably passed because there were no upfront error messages. yes, removing the packages from yaml file did not cause any errors. I thought that might be one possibility because the commit that did that caused the first failure. but now the checks are passing with no changes at all except just the change of removing the packages again from yaml file reverting to previous state which had caused failures earlier.